initrd/etc/ash_functions

```diff
@@ -80,9 +229,12 @@ recovery() {
 	fi
 	while [ true ]
 	do
+		#Going to recovery shell should be authenticated if supported
+                gpg_auth
+
```

Comment:
@JonathonHall-Purism so if reboot calls recovery which calls gpg_auth in current state of this PR, and recovery sources . /tmp/config, I would have expected gpg_auth to be testable prior of rebooting on qemu boards.

Reply:
fixedand commented at https://github.com/osresearch/heads/pull/1515#discussion_r1378137635